### DISCUSSION TITLE
looking for a dictionary kind structure for counting purposes

### DISCUSSION THREAD

**Discussion by alcyoneus-g** [reach-sh/reach-lang#1144] [Q&A] [answered]
Is there something like python's dictionary or HashMap in java kind of structure I could leverage in reach (algorand network)

What I plan to do is have the dictionary setup and there will be reach APIs that accept a type (the dictionary's key) and update the count values for the corresponding entry in the dictionary (value).

Something like:
```
{
   1: 3,
   2: 3,
   3: 4,
   ...
}
```
where the keys (1, 2, 3) are the categories/classes and the values are counts
and the key's length might be indefinite

thanks

**Comment by jeapostrophe** [comment, marked as answer]
https://docs.reach.sh/rsh/consensus/#mappings-creation-and-modification
